spec/unit/file_serving/metadata_spec.rb

```diff
@@ -18,8 +18,13 @@
     expect(Puppet::FileServing::Metadata.new(foobar)).to respond_to(:collect)
   end
 
-  it "should support pson serialization" do
-    expect(Puppet::FileServing::Metadata.new(foobar)).to respond_to(:to_pson)
+  it "should default to json" do
+    expect(Puppet::FileServing::Metadata.default_format).to eq(:json)
+  end
+
+  it "should support json, pson, yaml" do
+    # msgpack is optional, so using include instead of eq
+    expect(Puppet::FileServing::Metadata.supported_formats).to include(:json, :pson, :yaml)
```

Comment:
Just to make sure I understand - this support is the result of leveraging the indirector and defining the right `render` and `intern` methods? ie not necessarily explicitly specifying these in the class.